core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java

```diff
@@ -213,16 +213,22 @@ String formatStepText(String keyword, String stepText, Format textFormat, Format
         for (Argument argument : arguments) {
             // can be null if the argument is missing.
             if (argument.getOffset() != null) {
-                String text = stepText.substring(textStart, argument.getOffset());
-                result.append(textFormat.text(text));
+                int textEnd = argument.getOffset();
+                if (textStart > textEnd ) { // textStart can be > textEnd with nested capture groups
+                    continue;
+                } else {
```

Comment:
After a branching statement (`break`, `continue` and `return`) you can omit the else branch. It is implied.